pinot-common/src/main/java/org/apache/pinot/common/utils/PinotDataType.java

```diff
@@ -1119,13 +1090,68 @@ public PinotDataType getSingleValueType() {
   }
 
   public static PinotDataType getSingleValueType(Class<?> cls) {
-    PinotDataType pdt = SINGLE_VALUE_TYPE_MAP.get(cls);
-    return (pdt != null) ? pdt : OBJECT;
+    if (cls == String.class) {
```

Comment:
nit: maybe a brief comment that types are arranged per their frequency.

Reply:
Let's keep the supported storage type in the front because they are usually more commonly used, and more readable
`INT, LONG, FLOAT, DOUBLE, STRING, BYTES, BOOLEAN, TIMESTAMP, BYTE, CHARACTER, SHORT, OBJECT`